### PR TITLE
Collect telemetry metrics from Triton metrics endpoint

### PR DESCRIPTION
**Pull request by lkomali**
This PR is a part of adding telemetry metrics to GenAI-perf.

Changes implemented in this PR:
  1. Added a new Telemetry metrics class that has the GPU metrics mentioned below.
      - gpu_power_usage
      - gpu_power_limit
      - energy_consumption
      - gpu_utilization
      - total_gpu_memory
      - gpu_memory_used
  2. A new folder to hold all telemetry data collection files
  3. A Base Telemetry Data Collector class that implements a thread to gather metrics from endpoint.
  4. A specific Triton Telemetry Data Collector class that has a specific implementation for Triton metrics.
  5. Modified wrapper.py to start TelemtryDataCollector thread once perf_analyzer subprocess starts running.
  
 This is how the metrics are stored.
 
image

For every metric, the values are stored as a list of lists. 
The outer list represents the sequence of metric measurements over time.
Each inner list contains the metric values for each GPU at a particular time point.

### CODE REVIEW

File: genai-perf/genai_perf/constants.py [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
 DEFAULT_HTTP_URL = "localhost:8000"
 DEFAULT_GRPC_URL = "localhost:8001"
-
+DEFAULT_TRITON_METRICS_URL = "http://0.0.0.0:8002/metrics"
```
[the remark this answers is not in the document]
Comment (lkomali) [reply genai-perf/genai_perf/constants.py:29]: you are right, it should be localhost:8002/metrics. I modified it.

**Review by dyastremsky — DISMISSED** [review]
Excellent work, Harshini!

**Review by debermudez — APPROVED** [review]
Just wait to merge until @nv-hwoo closes his comments.
Nice work!

**Review by nv-hwoo — COMMENTED** [review]
Awesome work @lkomali! Thanks for adding such a detailed doc string 👍 And sorry for getting back so late 😅 All looks good with just one small comment about the optional argument for `TelemetryDataCollector` in the `run` function.

**Review by nv-hwoo — APPROVED** [review]

**Pull request merged**